### ISSUE TITLE
Using a smaller-sized image of RaspOS to create restore, bootup hangs on "random: crng init done"

### ISSUE THREAD

**Issue by mpegg4** [limepepper/ansible-role-raspberrypi#12]
Used an image of a RaspberryOS sd card with updated software / custom software to create the .restore.img  however on bootup constantly hung on "random: crng init done"

**Comment by tolland** [comment]
can you provide more details, and I'll try and replicate the problem. (need the base image, and details of what was added/changed etc)

that error usually shows up when the /boot/cmdline.txt doesn't match the actual IDs of the filesystems. (Also I've not tested it on the most recent RaspOSs) You might want to try with a fresh image, with no modifications and see if its something to do with how the RaspOS is setting up the partitions.

**Comment by mpegg4** [comment]
@tolland absolutely, will try fresh image first and report back thank you for the quick ack. amazing project btw thank you.

**Comment by mpegg4** [comment]
I think I am going to close this issue and spend more time studying your articles and exploring my source image. The images I tried were 1) Latest RaspbianOS and 
2) https://downloads.raspberrypi.org/raspbian/images/raspbian-2019-04-09/2019-04-08-raspbian-stretch.zip (2019 april stretch full).  
I wrote those images to SD card (16GB), then added some custom scripts/settings/configs for home use (wpa_supplicant , some access point code etc). 

After that I generated an .img from that home SD card but only using allocated bytes (so that it didnt create an .img that was 16GB+ since raspberry os tends to expand filesystem on firstboot).

That allocated-byte .img file is what i used for ./create-factory-reset but doesnt boot for image # 2) 

After using the script once I am also running into /dev/loop running out of space issues so I am thinking this is my poor or non-linux skills affecting the situation at this point ("`dd :  error writing '/dev/loop1p2': no space left on device`").

I notice specific setup in utils for 2020 buster lite image, so I will try to set that up as reset image first and go from there as a base image for my config needs.

thank you!
